xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/changesinline.vm

```diff
@@ -89,3 +93,8 @@
 #set ($nextNewdocVersion = $newdoc.documentArchive.getNextVersion($newdoc.RCSVersion))
 #template('changesdoc.vm')
 </div>
+## Pass the required skin extensions to the client side to be injected automatically in the HTML head by xwiki.js.
+#if ($xcontext.action == 'get')
+  #getRequiredSkinExtensions($requiredSkinExtensions)
+  #set ($discard = $response.setHeader('X-XWIKI-HTML-HEAD', $requiredSkinExtensions))
```

Comment:
Not a blocker for this PR, but I have the feeling that we should automate this more instead of having it in various tamplates. All it takes is to push an async context when an ajax request starts and automatically "augment" the response with the required HTTP headers when we are done executing it.

Reply:
Indeed. I'll create a jira issue for this